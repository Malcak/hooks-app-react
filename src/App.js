import React from 'react';
import { MemoHook } from './components/memos/MemoHook';
import { Memorize } from './components/memos/Memorize';
import { MultipleCumtomHooks } from './components/multipleCustomHooks/MultipleCumtomHooks';
import { FormApp } from './components/useEffect/FormApp';
import { FormWithCustomHook } from './components/useEffect/FormWithCustomHook';
import { LayoutApp } from './components/useLayoutEffect/LayoutApp';
import { FocusApp } from './components/useRef/FocusApp';
import { RealExampleRef } from './components/useRef/RealExampleRef';
import { CounterApp } from './components/useState/CounterApp';
import { CounterWithCustomHook } from './components/useState/CounterWithCustomHook';

export const App = () => {
  return (
    <div>
      <CounterApp />
      <hr />
      <CounterWithCustomHook />
      <hr />
      <FormApp />
      <hr />
      <FormWithCustomHook />
      <hr />
      <MultipleCumtomHooks />
      <hr />
      <FocusApp />
      <hr />
      <RealExampleRef />
      <hr />
      <LayoutApp />
      <hr />
      <Memorize />
      <hr />
      <MemoHook />Ã±
    </div>
  );
};
