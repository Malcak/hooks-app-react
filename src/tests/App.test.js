import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('tests on the <App /> component', () => {
  test('should display the component correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
