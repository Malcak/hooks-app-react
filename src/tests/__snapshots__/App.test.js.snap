// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <App /> component should display the component correctly 1`] = `
<div>
  <CounterApp />
  <hr />
  <CounterWithCustomHook />
  <hr />
  <FormApp />
  <hr />
  <FormWithCustomHook />
  <hr />
  <MultipleCumtomHooks />
  <hr />
  <FocusApp />
  <hr />
  <RealExampleRef />
  <hr />
  <LayoutApp />
  <hr />
  <Memorize />
  <hr />
  <MemoHook />
  <hr />
  <CallbackHook />
  <hr />
  <TodoApp />
  <hr />
  <MainApp />
</div>
`;
